docs/apis-tools/java-client/getting-started.md

```diff
@@ -0,0 +1,516 @@
+            .audience(AUDIENCE)
+            .clientId(CLIENT_ID)
+            .clientSecret(CLIENT_SECRET)
+            .keystorePath(KEYSTORE_PATH)
+            .keystorePassword(KEYSTORE_PASSWORD)
+            .keystoreKeyPassword(KEYSTORE_KEY_PASSWORD)
+            .truststorePath(TRUSTSTORE_PATH)
+            .truststorePassword(TRUSTSTORE_PASSWORD)
+            .build();
+
+    try (CamundaClient client = CamundaClient.newClientBuilder()
+            .grpcAddress(URI.create(CAMUNDA_GRPC_ADDRESS))
+            .restAddress(URI.create(CAMUNDA_REST_ADDRESS))
+            .credentialsProvider(credentialsProvider)
+            .build()) {
+
+        // Test the connection
+        client.newTopologyRequest().send().join();
+        System.out.println("Connected to Camunda 8!");
+    }
+}
+```
+
+**What this code does**
+
+1. **Sets up X.509 certificate authentication** – Configures the client to authenticate using client certificates with OAuth.
+2. **Builds a secure client** – Establishes an encrypted connection using mutual TLS authentication.
+3. **Connects to both APIs** – Configures access to the Zeebe gRPC and Orchestration Cluster REST APIs.
+4. **Tests the connection** – Verifies certificate authentication by requesting cluster topology information.
+
+**Environment variables option**  
+You can also set connection details via environment variables to create the client more simply:
+
+```bash
+export CAMUNDA_GRPC_ADDRESS='[Address of Zeebe API (gRPC) - default: http://localhost:26500]'
+export CAMUNDA_REST_ADDRESS='[Address of the Orchestration Cluster API - default: http://localhost:8080]'
+export CAMUNDA_OAUTH_URL='[OAuth URL e.g. http://localhost:18080/auth/realms/camunda-platform/protocol/openid-connect/token]'
+export CAMUNDA_TOKEN_AUDIENCE='[Audience - default: zeebe-api]'
+export CAMUNDA_CLIENT_ID='[Client ID]'
+export CAMUNDA_CLIENT_SECRET='[Client Secret]'
+export CAMUNDA_SSL_CLIENT_KEYSTORE_PATH='[Keystore path]'
+export CAMUNDA_SSL_CLIENT_KEYSTORE_SECRET='[Keystore password]'
+export CAMUNDA_SSL_CLIENT_KEYSTORE_KEY_SECRET='[Keystore material password]'
+export CAMUNDA_SSL_CLIENT_TRUSTSTORE_PATH='[Truststore path]'
+export CAMUNDA_SSL_CLIENT_TRUSTSTORE_SECRET='[Truststore password]'
+```
+
+```java
+CamundaClient client = CamundaClient.newClientBuilder().build();
+```
+
+The client automatically reads environment variables and configures the appropriate authentication method.
+
+Refer to your identity provider documentation for configuring X.509 authentication. For example, see [Keycloak](https://www.keycloak.org/server/mutual-tls).
+
+:::note
+
+- Ensure addresses use absolute URI format: `scheme://host(:port)`.
+- By default, environment variables override any values provided in Java code. To give Java code values precedence, use the `.applyEnvironmentOverrides(false)` method on `OAuthCredentialsProviderBuilder`.  
+  :::
+
+</TabItem>
+
+</Tabs>
+
+### Step 3: Start building your process application
```

Comment:
❓ we duplicate examples we have under `../java-client-examples` see http://localhost:3000/docs/next/apis-tools/java-client-examples/ which are also backed by an example repo. Let's align on the way forward (these are also outdated 😬 )

I would be in favor of removing them from now and rather start fresh. Covering deploy, start seems a good scope to get started, so I would be in favor of keeping this scope here.

We should extract the code to a repo though, where we can have some CI and rather embed the code files here.